MaxText/max_utils.py

```diff
@@ -100,6 +100,19 @@ def write_metrics_for_gcs(metrics, step, config, running_metrics):
     running_metrics = [] # reset running_metrics to empty list
   return running_metrics
 
+def write_train_config_for_gcs(config):
+  """Writes train config to gcs"""
+  train_config_filename = "train_config.txt"
+  with open(train_config_filename, 'w', encoding="utf8") as train_config_for_gcs:
```

Comment:
Nit: Is it possible to instead write the config in yaml format so it can be used in place of `configs/base.yml` for future runs? IMO this would slightly preferrable - although perhaps format is more convenient for downstream tasks you have in mind.

Reply:
I think `.txt` is fine, thanks for looking into it.

Reply:
Are we sure this is hard? What did you try?

Reply:
I changed it so that it outputs as a yml now. Here is an example of what the output yml looks like gpaste/5947072564428800. When it exports to yml, it uses a slightly different notation for specifying lists but I confirmed that it generates the same config when passed back into MaxText.